mobile/mysterium/wireguard_connection_setup.go

```diff
@@ -297,3 +300,25 @@ func (wg *wireguardConnection) runPeriodically(duration time.Duration) {
 		}
 	}
 }
+
+func (wg *wireguardConnection) waitHandshake() error {
+	for {
+		select {
+		case <-time.After(100 * time.Millisecond):
```

Comment:
```suggestion
		case <-time.After(20 * time.Millisecond):
```
These can be checked more frequently imho.